src/main.py

```diff
@@ -27,11 +27,13 @@
 from api import api_handlers
 
 # These should all inherit from base.handlers.BaseHandler
-_UNAUTHENTICATED_ROUTES = [#('/', handlers.RootHandler),
-                           ('/api/health', api_handlers.ApiHandler),
+_UNAUTHENTICATED_ROUTES = [('/api/health', api_handlers.ApiHandler),
                            ('/examples/xss', example_handlers.XssHandler),
                            ('/examples/xssi', example_handlers.XssiHandler)]
 
+# There should be only validators.MainPage to secure index.html
+_AUTHENTICATED_ROUTES = [('/.*', handlers.MainPage), ]
```

Comment:
Routes handled by `handlers.MainPage` should be specified explicitly. `webapp2` can then render a 404 page for any unmatched routes.